### ISSUE TITLE
Umlauts are not shown correctly in Alexa

### ISSUE THREAD

**Issue by stieler-it** [datech/node-red-contrib-amazon-echo#118]
##### SUMMARY
I create devices with umlaut (ä, ö, ü) and they are not shown corectly on Alexa web page.

##### ECHO DEVICES
* 1x Echo 2nd generation 
* 2x Echo Dot 3nd generation 

##### MODULE VERSION
```0.1.10```

##### CONFIGURATION
```
[{"id":"e32d9407.3d9368","type":"subflow","name":"Echo to Hue","info":"","category":"","in":[{"x":80,"y":140,"wires":[{"id":"2a7a5ad6.8c57b6"}]}],"out":[{"x":340,"y":140,"wires":[{"id":"2a7a5ad6.8c57b6","port":0}]}],"env":[],"color":"#DDAA99"},{"id":"d496c4a6.6cb0d8","type":"switch","z":"e32d9407.3d9368","name":"","property":"command","propertyType":"msg","rules":[{"t":"eq","v":"SetColorRequest","vt":"str"}],"checkall":"true","repair":false,"outputs":1,"x":170,"y":20,"wires":[["cd5710df.64125"]]},{"id":"cd5710df.64125","type":"change","z":"e32d9407.3d9368","name":"{\"hue\":0,\"saturation\":1,\"brightness\":1} to [h,s,v]","rules":[{"t":"set","p":"payload","pt":"msg","to":"[ payload.hue, payload.saturation * 100,\tpayload.brightness * 100 ]","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":400,"y":80,"wires":[["c2e399dd.b9ba38"]]},{"id":"c2e399dd.b9ba38","type":"color-convert","z":"e32d9407.3d9368","input":"hsv","output":"rgb","outputType":"array","scaleInput":false,"x":670,"y":80,"wires":[["39397c2.d602484"]]},{"id":"39397c2.d602484","type":"change","z":"e32d9407.3d9368","name":"Hue Color Change","rules":[{"t":"set","p":"payload","pt":"msg","to":"{ \"rgb\": payload }","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":830,"y":40,"wires":[[]]},{"id":"2a7a5ad6.8c57b6","type":"change","z":"e32d9407.3d9368","name":"Echo -> HUE","rules":[{"t":"set","p":"payload","pt":"msg","to":"{ \"on\": msg.on, \"brightnessLevel\": msg.bri, \"rgb\": msg.rgb }","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":210,"y":140,"wires":[[]]},{"id":"c3c8c242.8afc9","type":"tab","label":"Alexa","disabled":false,"info":""},{"id":"1b86f27b.837bee","type":"subflow:e32d9407.3d9368","z":"c3c8c242.8afc9","name":"","x":570,"y":80,"wires":[["58515f9f.4cddc"]]},{"id":"36dca496.14e7ec","type":"amazon-echo-hub","z":"c3c8c242.8afc9","port":"1080","processinput":0,"discovery":true,"x":130,"y":220,"wires":[["34bce434.988c0c","e5e17dcf.1c972","4db324db.f5f6dc","dd196fc3.7cc29"]]},{"id":"34bce434.988c0c","type":"amazon-echo-device","z":"c3c8c242.8afc9","name":"Wohnzimmer Licht","topic":"","x":370,"y":80,"wires":[["1b86f27b.837bee"]]},{"id":"e5e17dcf.1c972","type":"amazon-echo-device","z":"c3c8c242.8afc9","name":"Esszimmer Licht","topic":"","x":360,"y":120,"wires":[["c5f39c29.96e6e"]]},{"id":"c5f39c29.96e6e","type":"subflow:e32d9407.3d9368","z":"c3c8c242.8afc9","name":"","x":570,"y":120,"wires":[["6e32415e.4510f"]]},{"id":"4db324db.f5f6dc","type":"amazon-echo-device","z":"c3c8c242.8afc9","name":"Küchen Licht","topic":"","x":350,"y":160,"wires":[["a74c105e.728d2"]]},{"id":"a74c105e.728d2","type":"subflow:e32d9407.3d9368","z":"c3c8c242.8afc9","name":"","x":570,"y":160,"wires":[["f7e1d3d7.d4dc5"]]},{"id":"dd196fc3.7cc29","type":"amazon-echo-device","z":"c3c8c242.8afc9","name":"Büro unten Licht","topic":"","x":360,"y":220,"wires":[["5db34094.d81c3"]]},{"id":"5db34094.d81c3","type":"subflow:e32d9407.3d9368","z":"c3c8c242.8afc9","name":"","x":570,"y":220,"wires":[["7aea90a8.58fd3"]]},{"id":"58515f9f.4cddc","type":"link out","z":"c3c8c242.8afc9","name":"Wohnzimmer Licht","links":[],"x":735,"y":80,"wires":[]},{"id":"6e32415e.4510f","type":"link out","z":"c3c8c242.8afc9","name":"Esszimmer Licht","links":[],"x":735,"y":120,"wires":[]},{"id":"f7e1d3d7.d4dc5","type":"link out","z":"c3c8c242.8afc9","name":"Küchen Licht","links":[],"x":735,"y":160,"wires":[]},{"id":"7aea90a8.58fd3","type":"link out","z":"c3c8c242.8afc9","name":"Büro unten Licht","links":["334e3316.aba04c"],"x":735,"y":220,"wires":[]}]
```

##### CONSOLE OUTPUT
n/a

##### STEPS TO REPRODUCE
1. Create a hub node
2. Create a device with a name with umlaut
3. Discover devices

##### EXPECTED RESULTS
Devices are shown as entered.

##### ACTUAL RESULTS
Devices are shown but umlauts are broken, probably due to a wrong encoding.

**Comment by stieler-it** [comment]
Strangely, if I write "Kuechen Licht" in the flow it seems to be recognized as "Küchen Licht" in Alexa web app.

**Comment by datech** [comment]
Duplicate to #74 

Is the "Kuechen - Küchen" workaround working in other cases?

**Comment by stieler-it** [comment]
Sorry, I only looked through the open tickets before posting. This issue explains that is's a problem of some Amazon devices. Who is changing "Kuechen" to "Küchen" on the way?

**Comment by datech** [comment]
> Strangely, if I write "Kuechen Licht" in the flow it seems to be recognized as "Küchen Licht" in Alexa web app.

I was just wondering, will Alexa app recognize other words correctly if you replace "ü" with "ue".

**Comment by stieler-it** [comment]
You are talking about voice recognition, right? But for some reason already the text is converted to contain Umlauts. Like a workaround would be in place somewhere.

**Comment by datech** [comment]
Yep, a few months ago, Alexa just stopped displaying umlauts correctly. Obviously, it is not fixed by Amazon, and it is still not working on some devices.

**Comment by stieler-it** [comment]
I noticed that there is a subtle different in the response of this software in comparison to my Hue hub. This app returns a Content-Type with `;charset=utf-8`, charset information is missing in the real hub.

But, I checked the responses and both **do** send utf-8 encoded strings.

However, the strange thing is, even if I now use umlauts again in my device descriptions I cannot reproduce the bug again. Names are shown correctly in Alexa web app - both from this software and from Hue hub.

In any case this is duplicate and thus can be closed. Thanks for your support.
